js/vm/typeaheadlist.js

```diff
@@ -26,20 +26,11 @@ export default class LectureList {
     }
   };
 
-  filtered() {
-    if (this.searchString === '') {
-      return this;
-    }
-    let regex = this.getSearchRegex(this.searchString);
-
-    return this.lectures.filter(l => regex.test(l.name));
-  }
-
   get typeaheadDataset() {
```

Comment:
Now that the dataset isn't actually used anymore except for its `source` method, I think you can just delete it and inline the `source` implementation at the call site in `DocumentSelection`.